py/torch_tensorrt/dynamo/conversion/impl/pad.py

```diff
@@ -18,39 +20,109 @@
 """
 
 
-def constant_padNd(
+def get_padded_shape_tensors(
     ctx: ConversionContext,
     target: Union[Target, str],
     source_ir: Optional[SourceIR],
     name: str,
     input: TRTTensor,
     pad: Sequence[int],
-    value: Union[int, float] = 0,
 ) -> TRTTensor:
-    if has_dynamic_shape(input.shape):
-        assert input.shape[1] != -1, "Channel dim can't be dynamic for padding."
-
     rank = len(input.shape)
-
     if len(pad) // 2 > rank:
         raise RuntimeError(
-            f"Trying to pad last {len(pad) // 2} dimension but the input only has {rank} dimension."
+            f"Trying to pad last {len(pad) // 2} dimensions but the input only has {rank} dimensions."
         )
 
+    input_shape_tensor = get_shape_with_dynamic_shape(
+        ctx,
+        target,
+        source_ir,
+        name + "_input_shape",
+        input.shape,
+        input,
+    )
+    padded_shape_tensor = input_shape_tensor
+
     start_list = [0] * rank
-    new_shape = list(input.shape)
+    for i in range(len(pad) // 2):
+        dim_index = rank - (i + 1)
+        pad_before = pad[i * 2]
+        pad_after = pad[i * 2 + 1]
 
-    for i in range(0, len(pad) // 2):
-        start_list[-i - 1] = -pad[i * 2]
-        new_shape[-i - 1] += pad[i * 2] + pad[i * 2 + 1]
+        pad_sum = get_trt_tensor(
+            ctx, pad_before + pad_after, f"{name}_pad_sum_{i}", dtype=np.int32
+        )
+        dim_shape = ctx.net.add_slice(
+            input_shape_tensor,
+            start=(dim_index,),
+            shape=(1,),
+            stride=(1,),
+        ).get_output(0)
+
+        new_dim_shape = impl.elementwise.add(
+            ctx, target, source_ir, f"{name}_shape_dim_{i}", dim_shape, pad_sum
+        )
+        start_list[dim_index] = -pad_before
```

Comment:
Curious : Why are the values negative here and what is the expectation of TensorRT when you do `layer.set_input(1, start_indices_tensor)` later ?

Reply:
The negative values in `start_list[dim_index] = -pad_before` are used to indicate the starting point for the slicing operation in TensorRT's `ISliceLayer`. When setting `layer.set_input(1, start_indices_tensor)`, TensorRT interprets these negative start indices to appropriately offset the input tensor before applying padding.